### PR TITLE
Add release notes for 7.10.0

### PR DESCRIPTION
**Pull request by andreidan**
## Preview

https://elasticsearch_63460.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.10/release-notes-7.10.0.html

### CODE REVIEW

File: docs/reference/release-notes/7.10.asciidoc [hunk 1]
```diff
@@ -1,15 +1,279 @@
...
+Infra/Core::
+* Add logstash system index APIs {es-pull}53350[#53350]
+* Deprecate REST access to System Indices {es-pull}60945[#60945]
+* Speed up Compression Logic by Pooling Resources {es-pull}61358[#61358]
+* System index reads in separate threadpool {es-pull}57936[#57936] (issues: {es-issue}37867[#37867], {es-issue}50251[#50251])
+
+Infra/Logging::
+* Do not create two loggers for DeprecationLogger {es-pull}58435[#58435]
+* Header warning logging refactoring {es-pull}55941[#55941] (issues: {es-issue}52369[#52369], {es-issue}55699[#55699])
+* Write deprecation logs to a data stream {es-pull}58924[#58924] (issue: {es-issue}46106[#46106])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:133]: It looks like this PR was closed but not merged.
```suggestion
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 2]
```diff
@@ -1,15 +1,279 @@
...
+Infra/Packaging::
+* Add UBI docker builds {es-pull}60742[#60742]
+* Upgrade Centos version in Dockerfile to 8 {es-pull}59019[#59019]
+
+Infra/Resiliency::
+* Remove node from cluster when node locks broken {es-pull}61400[#61400] (issues: {es-issue}52680[#52680], {es-issue}58373[#58373])
+
+Infra/Scripting::
+* Scripting: Augment String with sha1 and sha256 {es-pull}59671[#59671] (issue: {es-issue}59633[#59633])
+* Scripting: Converts casting and def support {es-pull}61350[#61350] (issue: {es-issue}59647[#59647])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:145]: I don't think we need these with the above label.
```suggestion
* Augment String with sha1 and sha256 {es-pull}59671[#59671] (issue: {es-issue}59633[#59633])
* Converts casting and def support {es-pull}61350[#61350] (issue: {es-issue}59647[#59647])
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 3]
```diff
@@ -1,15 +1,279 @@
...
+
+Machine Learning::
+* Add a "verbose" option to the data frame analytics stats endpoint {es-pull}59589[#59589] (issue: {es-issue}59125[#59125])
+* Add new include flag to GET inference/<model_id> API for model training metadata {es-pull}61922[#61922]
+* Adds new feature_processors field for data frame analytics {es-pull}60528[#60528] (issue: {es-issue}59327[#59327])
+* Adds new n_gram_encoding custom processor {es-pull}61578[#61578]
+* Create new dynamic domain feature extractor processor {es-pull}61874[#61874]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:152]: This looks to be closed but not merged.
```suggestion
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 4]
```diff
@@ -1,15 +1,279 @@
...
+* Log and track open/close of transport connections {es-pull}60297[#60297]
+
+Performance::
+* Speed up empty highlighting many fields {es-pull}61860[#61860]
+
+SQL::
+* Add option to provide the delimiter for the CSV format {es-pull}59907[#59907] (issue: {es-issue}41634[#41634])
+* Implement DATE_PARSE function for parsing strings into DATE values {es-pull}57391[#57391] (issue: {es-issue}54962[#54962])
+* Implement FORMAT function {es-pull}55454[#55454] (issue: {es-issue}54965[#54965])
+* QL: Wildcard support {es-pull}59753[#59753]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:172]: This is an issue, not a PR.
```suggestion
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 5]
```diff
@@ -1,15 +1,279 @@
...
+
+Search::
+* Avoid reloading _source for every inner hit. {es-pull}60494[#60494] (issue: {es-issue}32818[#32818])
+* Cancel multisearch when http connection closed {es-pull}61399[#61399]
+* Enable cancellation for msearch requests {es-pull}61337[#61337]
+* Executes incremental reduce in the search thread pool {es-pull}58461[#58461] (issues: {es-issue}51857[#51857], {es-issue}53411[#53411])
+* Introduce point in time APIs in x-pack basic {es-pull}61062[#61062] (issues: {es-issue}26472[#26472], {es-issue}46523[#46523])
+* Make multi search tasks cancellable {es-pull}44494[#44494]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:180]: This looks to be closed but not merged.
```suggestion
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 6]
```diff
@@ -1,15 +1,279 @@
...
+* ParametrizedFieldMapper to run validators against default value {es-pull}60042[#60042] (issue: {es-issue}59332[#59332])
+* Search - add case insensitive flag for "term" family of queries {es-pull}61596[#61596] (issue: {es-issue}61546[#61546])
+* Search - add case insensitive support for regex queries. {es-pull}59441[#59441]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:183]: I'd remove duplicate labels and unneeded punctuation.
```suggestion
* Add case insensitive flag for "term" family of queries {es-pull}61596[#61596] (issue: {es-issue}61546[#61546])
* Add case insensitive support for regex queries {es-pull}59441[#59441]
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 7]
```diff
@@ -1,15 +1,279 @@
...
+* Clone Snapshot API {es-pull}61839[#61839]
+* Determine shard size before allocating shards recovering from snapshots {es-pull}61906[#61906]
+* Introduce index based snapshot blob cache for Searchable Snapshots {es-pull}60522[#60522]
+* Validate snapshot UUID during restore {es-pull}59601[#59601] (issue: {es-issue}50999[#50999])
+
+Store::
+* Report more details of unobtainable ShardLock {es-pull}61255[#61255] (issue: {es-issue}38807[#38807])
+
+Task Management::
+* Support task cancellation cross clusters {es-pull}55779[#55779]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:197]: This looks to be closed but not merged.
```suggestion
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 8]
```diff
@@ -1,15 +1,279 @@
...
+* Fix cluster health when closing {es-pull}61709[#61709]
+
+Engine::
+* Fix estimate size of translog operations {es-pull}59206[#59206]
+
+Features/ILM+SLM::
+* Fix ILM history index settings {es-pull}61880[#61880] (issues: {es-issue}61457[#61457], {es-issue}61863[#61863])
+
+Features/Java Low Level REST Client::
+* [client] Handle non-default port in Cloud-Id {es-pull}61581[#61581]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:235]: I don't think the label is needed here.
```suggestion
* Handle non-default port in Cloud-Id {es-pull}61581[#61581]
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 9]
```diff
@@ -1,15 +1,279 @@
...
+* Improve 'ignore_malformed' handling for dates {es-pull}60211[#60211] (issue: {es-issue}52634[#52634])
+
+Network::
+* Let `isInetAddress` utility understand the scope ID on ipv6 {es-pull}60172[#60172] (issue: {es-issue}60115[#60115])
+* Suppress noisy SSL exceptions {es-pull}61359[#61359]
+
+Search::
+* Allows nanosecond resolution in search_after {es-pull}60328[#60328] (issue: {es-issue}52424[#52424])
+* Consolidate validation for 'docvalue_fields'. {es-pull}59473[#59473]
+* Correct how field retrieval handles multifields and copy_to. {es-pull}61309[#61309] (issue: {es-issue}61033[#61033])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:270]: The punctuation isn't needed here.
```suggestion
* Consolidate validation for 'docvalue_fields' {es-pull}59473[#59473]
* Correct how field retrieval handles multifields and copy_to {es-pull}61309[#61309] (issue: {es-issue}61033[#61033])
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 10]
```diff
@@ -1,15 +1,279 @@
...
+* In field retrieval, skip values that can't be parsed. {es-pull}60943[#60943]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:271]: This looks to be closed but not merged.
```suggestion
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 11]
```diff
@@ -1,15 +1,279 @@
...
+* ILM: add force-merge step to searchable snapshots action {es-pull}60819[#60819] (issues: {es-issue}53488[#53488], {es-issue}56215[#56215])
+
+Machine Learning::
+* Implement AucRoc metric for classification {es-pull}60502[#60502] (issue: {es-issue}62160[#62160])
+
+Mapping::
+* Introduce 64-bit unsigned long field type {es-pull}60050[#60050] (issue: {es-issue}32434[#32434])
+
+Search::
+* Add search 'fields' option to support high-level field retrieval. {es-pull}60100[#60100] (issues: {es-issue}49028[#49028], {es-issue}55363[#55363])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:63]: The punctuation isn't needed here.
```suggestion
* Add search 'fields' option to support high-level field retrieval {es-pull}60100[#60100] (issues: {es-issue}49028[#49028], {es-issue}55363[#55363])
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 12]
```diff
@@ -1,15 +1,279 @@
...
+* Add more useful toString on cluster state observers {es-pull}60277[#60277]
+* Fail invalid incremental cluster state writes {es-pull}61030[#61030]
+* Provide option to allow writes when master is down {es-pull}60605[#60605]
+
+Distributed::
+* Detect noop of update index settings {es-pull}61348[#61348]
+* Thread safe clean up of LocalNodeModeListeners {es-pull}59932[#59932] (issue: {es-issue}59801[#59801])
+
+Features/CAT APIs::
+* Adding Hit counts and Miss counts for QueryCache exposed through REST api. This resolves:#48645 {es-pull}60114[#60114] (issue: {es-issue}48645[#48645])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:104]:
```suggestion
* Adding Hit counts and Miss counts for QueryCache exposed through REST API {es-pull}60114[#60114] (issue: {es-issue}48645[#48645])
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 13]
```diff
@@ -1,15 +1,279 @@
...
+* Speed up empty highlighting many fields {es-pull}61860[#61860]
+
+SQL::
+* Add option to provide the delimiter for the CSV format {es-pull}59907[#59907] (issue: {es-issue}41634[#41634])
+* Implement DATE_PARSE function for parsing strings into DATE values {es-pull}57391[#57391] (issue: {es-issue}54962[#54962])
+* Implement FORMAT function {es-pull}55454[#55454] (issue: {es-issue}54965[#54965])
+* QL: Wildcard support {es-pull}59753[#59753]
+
+Search::
+* Avoid reloading _source for every inner hit. {es-pull}60494[#60494] (issue: {es-issue}32818[#32818])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.10.asciidoc:175]: The punctuation isn't needed here.
```suggestion
* Avoid reloading _source for every inner hit {es-pull}60494[#60494] (issue: {es-issue}32818[#32818])
```

**Review by jrodewig — APPROVED** [review]
Looks good overall, but I left a few suggestions. It looks like there are some PRs listed that were not merged.

I'll also defer to @lcawl on the ML-related changes. Thanks @andreidan!

**Comment by andreidan** [comment]
@jrodewig thanks for the review!

**Comment by andreidan** [comment]
@lcawl would you like to also have a look at the machine learning content?

**Comment by lcawl** [comment]
I added content from https://github.com/elastic/ml-cpp/blob/7.10/docs/CHANGELOG.asciidoc and edited the machine learning PRs.

**Review by lcawl — APPROVED** [review]

File: docs/reference/release-notes/7.10.asciidoc [hunk 14]
```diff
@@ -1,15 +1,276 @@
...
+Infra/Scripting::
+* Augment String with sha1 and sha256 {es-pull}59671[#59671] (issue: {es-issue}59633[#59633])
+* Converts casting and def support {es-pull}61350[#61350] (issue: {es-issue}59647[#59647])
+
+Machine Learning::
+* Add a "verbose" option to the data frame analytics stats endpoint {es-pull}59589[#59589] (issue: {es-issue}59125[#59125])
+* Add new include flag to get trained models API for model training metadata {es-pull}61922[#61922]
+* Add new feature_processors field for data frame analytics {es-pull}60528[#60528] (issue: {es-issue}59327[#59327])
+* Add new n_gram_encoding custom processor {es-pull}61578[#61578]
+* Make MlDailyMaintenanceService delete jobs that are in deleting state {es-pull}60121[#60121] (issue: {es-issue}42840[#42840])
```
Comment (droberts195) [inline docs/reference/release-notes/7.10.asciidoc:151]:
```suggestion
* Delete jobs whose deletion tasks were lost during nightly maintenance {es-pull}60121[#60121] (issue: {es-issue}42840[#42840])
```

File: docs/reference/release-notes/7.10.asciidoc [hunk 15]
```diff
@@ -1,15 +1,276 @@
...
+* Respect ML upgrade mode in TrainedModelStatsService {es-pull}61143[#61143]
```
Comment (droberts195) [inline docs/reference/release-notes/7.10.asciidoc:152]:
```suggestion
* Suspend persistence of trained model stats when ML upgrade mode is enabled {es-pull}61143[#61143]
```

**Review by droberts195 — COMMENTED** [review]
A couple of tweaks to the ML items to replace internal class names with things users are more likely to relate to

**Pull request merged**